src/test/java/io/stargate/sgv2/jsonapi/api/v1/FindOneAndReplaceIntegrationTest.java

```diff
@@ -275,6 +275,61 @@ public void withSort() {
           .body("data.docs[0]", jsonEquals(expected));
     }
 
+    @Test
+    public void withUpsert() {
+      String expected =
+          """
+        {
+          "_id": "doc2",
+          "username": "username2",
+          "status" : true
+        }
+        """;
+
+      String json =
+          """
+        {
+          "findOneAndReplace": {
+            "filter" : {"_id" : "doc2"},
```

Comment:
I'd recommend testing that additional filter properties don't end up in the document, like `"filter": {"_id" : "doc2", "otherProp": 44}` wouldn't end up with "otherProp" in the final document.